impl/impl.go

```diff
@@ -279,7 +273,7 @@ func (m *manager) SendVoucher(ctx context.Context, channelID datatransfer.Channe
 		span.SetStatus(codes.Error, err.Error())
 		return err
 	}
-	updateRequest, err := message.VoucherRequest(channelID.ID, voucher.Type(), voucher)
+	updateRequest, err := message.VoucherRequest(channelID.ID, chst.VoucherType(), voucher)
```

Comment:
This is an example that will fall down in the context of actual retrieval -- these types may not be the same. On the other hand, based on our modified markets code, it probably does not matter that much.